python/mxnet/gluon/data/dataset.py

```diff
@@ -135,6 +170,38 @@ def __call__(self, x, *args):
             return (self._fn(x),) + args
         return self._fn(x)
 
+class _FilteredDataset(Dataset):
+    """Dataset with a filter applied"""
+    def __init__(self, dataset, fn):
+        self._dataset = dataset
+        self._indices = []
```

Comment:
`self._indices = [i for for i, sample in enumerate(dataset) if fn(sample)]`
should be faster for large datasets.

Reply:
Good point I'll update it

Reply:
Sorry there was a typo in above code sample. You need to remove a `for`